src/jdenticonplugin.cpp

```diff
@@ -4,7 +4,12 @@
 QString
 JdenticonPlugin::generate(const QString &message, uint16_t size)
 {
+    JdenticonInfo info{ message, size };
+    if (m_cachedJdenticons.contains(info))
+        return m_cachedJdenticons[info];
+
     QByteArray byteArr = QByteArray::fromStdString(message.toStdString());
     Identicon identicon(byteArr, size);
-    return identicon.generateSvg(identicon, false);
+    m_cachedJdenticons[info] = identicon.generateSvg(identicon, false);
+    return m_cachedJdenticons[info];
```

Comment:
Imo the application should control the cache size. Alternatively use a cache with a bounded size.